### DISCUSSION TITLE
Creating Custom Filaments - Looking for clarification

### DISCUSSION THREAD

**Discussion by adminmat** [OrcaSlicer/OrcaSlicer#4193] [General]
Hello, I am hoping to understand better the process of creating custom filaments. As I understand there are two ways. One is by editing an existing filament in the left panel, and saving with a new name. Then there is this option: Clicking on the cog in the filament panel, then the Custom Filament tab:

<img width="956" alt="Settings" src="https://github.com/SoftFever/OrcaSlicer/assets/64502231/96c5d5aa-f29a-409f-9c09-2189d5bc5ac9">

Then "Create New"

<img width="611" alt="settings 2" src="https://github.com/SoftFever/OrcaSlicer/assets/64502231/695eb617-e943-475a-b92c-c3e1d5452840">

The second method seems to behave differently and I'm trying to understand when you would use this method. 

I do not understand what theses options mean such as: "Add filament preset under this filament"

<img width="362" alt="Screenshot 2024-02-22 150253" src="https://github.com/SoftFever/OrcaSlicer/assets/64502231/deb74eec-26b7-4fbf-bb5f-05d5f6c05447">

I have searched and don't see an explanation in the documentation. Most people on discord and the forums avoid this method but cant explain why. Thanks.

**Comment by dharahs14** [comment]
this is the same problem i am facing. i am not able to add other filament (Like ASA), even if it is added, i am not able to select it in the filament option!. even the filament is shown in SYSTEM FILAMENTS, i cannot select it in the filament option

**Reply by lin-ycv** [reply]
your issue is different, OP is asking what the custom filament feature is, you're asking why you can't add a filament.
to answer your question, it's most likely because of the nozzle setting you have for your printer. orca check if a filament is compatible with a filament type, if it's not, it won't show that filament as an option.
you proably has the nozzle set to brass, that's why ASA, an abrasive filament, is not shown.

**Comment by lin-ycv** [comment]
@adminmat not sure if you've found the answer, but from what i understand, there's no real difference, just different ways of doing the same thing.
the first method allows you to modify all the settings, then save it as a profile.
whereas the second method create a duplicate first, then you change the settings.
second method is obviously more cumbersome, because you have to navigate thru more UI's
the one benefit i see of the second method is that you can duplicate from a preset file, so it doesn't have to a profile that you have already loaded.

**Comment by kleinwareio** [comment]
This doesn't seem to be documented anywhere, but creating a new filament via the "cog", which now looks like a slider icon, creates the filament as a "base" filament. It is stored (`Menu -> Help -> Show Configuration Folder`) in a subfolder of your normal user filaments which is the only way I can tell that it's identified as different.

![image](https://github.com/user-attachments/assets/d17f123c-7708-4ecf-8b93-71aa3f7ae14b)

![image](https://github.com/user-attachments/assets/b53e4241-179f-43cf-a05e-061d96751eed)


Once you've created your base filament, you can then select it, edit it, and save as a new filament like you mentioned as the first technique. This appears to do the same thing, but it doesn't. The new filament that was created by editing your new base filament will now "inherit" from the base filament. This means you can edit properties on the base filament and it will be reflected in all the filaments that have inherited from it (unless they've explicitly set their own value).

You can go back to the "config" and click the "Edit" icon to see what filaments are inheriting from this base filament.

![image](https://github.com/user-attachments/assets/aa4173fe-d6b7-4865-8d13-3b89ce984372)

![image](https://github.com/user-attachments/assets/b078fa8c-9660-4c7a-8560-1f81a014c3f6)

I'm currently using this to specify different colors while sharing the base attribute from the base "Custom Filament".

**Comment by steadmanc** [comment]
I go through the process but it keeps failing.
![image](https://github.com/user-attachments/assets/2e1bd4b9-2a90-4f10-96d8-4164e0752bb3)
![image](https://github.com/user-attachments/assets/e77972c8-0704-4a93-8cb4-483935fc9ed1)
Tried both methods, Create Based on Current Filament and Copy Current Filament Preset.  Both give the "You have not selected a printer or preset yet.  Please select at least one." error.  Unlike in a video on how to add it, I never see a printer to select after "We could create the filament presets for your following printer:".

**Reply by mjparme** [reply]
I have the same problem. I continue to be amazed that OrcaSlicer is popular. You can't even create filaments. What does this error even mean? It doens't give me the oppurtunity to select a printer.

<img width="579" height="583" alt="Screenshot 2026-01-23 at 3 14 11 PM" src="https://github.com/user-attachments/assets/34b0abb6-308e-4f79-8ed8-09ba64296da7" />
